pkg/occlient/occlient.go

```diff
@@ -1116,23 +1119,44 @@ func (c *Client) WaitAndGetPod(selector string) (*corev1.Pod, error) {
 		return nil, errors.Wrapf(err, "unable to watch pod")
 	}
 	defer w.Stop()
-	for {
-		val, ok := <-w.ResultChan()
-		if !ok {
-			break
-		}
-		if e, ok := val.Object.(*corev1.Pod); ok {
-			glog.V(4).Infof("Status of %s pod is %s", e.Name, e.Status.Phase)
-			switch e.Status.Phase {
-			case corev1.PodRunning:
-				glog.V(4).Infof("Pod %s is running.", e.Name)
-				return e, nil
-			case corev1.PodFailed, corev1.PodUnknown:
-				return nil, errors.Errorf("pod %s status %s", e.Name, e.Status.Phase)
+
+	podChannel := make(chan *corev1.Pod)
+	watchErrorChannel := make(chan error)
+
+	go func() {
+	loop:
```

Comment:
I understand the requirement of this is to jump directly out from within the switch and for but because the labels hold the potential to make the code cryptic I personally have a strong dislike for them :(

But feel free to ignore it because labels are also an integral part of code ;) and they deserve equality ;)

Reply:
I have the same opinion as well... I really don't like the label but I think that the alternative of having various control flags would make things even weirder.

I am totally on the fence with this, don't really know what to do :P